### PR TITLE
Sql component should save db object on Init instead of creating a new

### PR DESCRIPTION
**Pull request by LMWF**
# Description

Save the db object during Init() so it can be used for operations - don't recreate it every time.

Also fixes an issue when using grpc - the data should not be marshalled if it's already []byte.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #397, #398

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by youngbupark** [comment]
What happens if db client is disconnected? In this case, reconnection is required. So I would create helper method to check if existing client connection is valid first and then use the existing connection. otherwise, I would reconnect it first.

**Comment by LMWF** [comment]
I don't think the user needs to manage that themselves:

https://golang.org/pkg/database/sql/#Open
```
The returned DB is safe for concurrent use by multiple goroutines and maintains its own pool of idle connections. Thus, the Open function should be called just once. It is rarely necessary to close a DB.
```

Also it looks like everything internally calls the conn() method which will manage connections for you:
https://golang.org/src/database/sql/sql.go?s=46419:46453#L1133

**Comment by youngbupark** [comment]
> I don't think the user needs to manage that themselves:
> 
> https://golang.org/pkg/database/sql/#Open
> 
> ```
> The returned DB is safe for concurrent use by multiple goroutines and maintains its own pool of idle connections. Thus, the Open function should be called just once. It is rarely necessary to close a DB.
> ```
> 
> Also it looks like everything internally calls the conn() method which will manage connections for you:
> https://golang.org/src/database/sql/sql.go?s=46419:46453#L1133

if the client has reconnection feature by itself, then we do not need to care the connection status.

File: state/sqlserver/sqlserver.go [hunk 1]
```diff
@@ -478,9 +450,16 @@ type dbExecutor interface {
 }
 
 func (s *SQLServer) executeSet(db dbExecutor, req *state.SetRequest) error {
-	json, err := json.Marshal(req.Value)
-	if err != nil {
-		return err
+	var err error
```
Comment (youngbupark) [inline state/sqlserver/sqlserver.go:453]: is this same behavior of redis component?

Comment (LMWF) [reply state/sqlserver/sqlserver.go:453]: Yes.  Other components may have a similar issue as well when handing data received by the grpc api.

**Comment by yaron2** [comment]
lgtm

**Pull request merged**